### PR TITLE
[DF-110] test_CheckAddingNewItemToTables [Elements/WebTables]

### PR DESCRIPTION
**Pull request by yuriiormson**
https://fem22.atlassian.net/browse/DF-110

### CODE REVIEW

**Review by DimaZadrutsiy — APPROVED** [review]

File: src/test/java/pages/BasePage.java [hunk 1]
```diff
@@ -428,5 +438,83 @@ protected boolean isAlertDisplayed() {
             return false;
         }
     }
+
+    protected String getSaltString10() {
+        String SALTCHARS = "AaBbCcDdEeFfGgHhIiJjKkLlMmNnOoPpQqRrSsTtUuVvWwXxYyZz";
```
Comment (KristinaPereselkina) [inline src/test/java/pages/BasePage.java:443]: String SALTCHARS = change to regex

File: src/test/java/pages/BasePage.java [hunk 2]
```diff
@@ -428,5 +438,83 @@ protected boolean isAlertDisplayed() {
...
+            int index = (int) (rnd.nextFloat() * SALTCHARS.length());
+            salt.append(SALTCHARS.charAt(index));
+        }
+        String saltStr = salt.toString();
+        return saltStr;
+
+    }
+
+    public String getSaltStringLettersNumbers10() {
+        String SALTCHARS = "AaBbCcDdEeFfGgHhIiJjKkLlMmNnOoPpQqRrSsTtUuVvWwXxYyZz1234567890";
```
Comment (KristinaPereselkina) [inline src/test/java/pages/BasePage.java:495]: change to regex

File: src/test/java/pages/BasePage.java [hunk 3]
```diff
@@ -428,5 +438,83 @@ protected boolean isAlertDisplayed() {
...
+            int index = (int) (rnd.nextFloat() * SALTCHARS.length());
+            salt.append(SALTCHARS.charAt(index));
+        }
+        String saltStr = salt.toString();
+        return saltStr;
+
+    }
+
+    protected String getSaltStringNumbersSymbols10() {
+        String SALTCHARS = "AaBbCcDdEeFfGgHhIiJjKkLlMmNnOoPpQqRrSsTtUuVvWwXxYyZz1234567890±!@#$%^&*()_+=-§";
```
Comment (KristinaPereselkina) [inline src/test/java/pages/BasePage.java:508]: change to regex

File: src/test/java/pages/BasePage.java (hunk 1 above)
Comment (KristinaPereselkina) [inline src/test/java/pages/BasePage.java:443]: specify name of the string SALTCHARS? what does it means

**Review by KristinaPereselkina — CHANGES_REQUESTED** [review]
Why do we need a lot of methods in base test? they are almost identical inside. Could we combine them in 1 general method usinf if , else if?

File: src/test/java/pages/BasePage.java (hunk 1 above)
Comment (yuriiormson) [reply src/test/java/pages/BasePage.java:443]: Removed couple of unused methods with SaltString

Comment (yuriiormson) [reply src/test/java/pages/BasePage.java:443]: I don't know how to make it. If you have proposition I'm opened for it)

Comment (yuriiormson) [reply src/test/java/pages/BasePage.java:443]: I don`t understand what you want)

Comment (yuriiormson) [reply src/test/java/pages/BasePage.java:443]: SALTCHARS it means that there is different letters

Comment (yuriiormson) [reply src/test/java/pages/BasePage.java]: @KristinaPereselkina check new PR)

**Comment by yuriiormson** [comment]
> Why do we need a lot of methods in base test? they are almost identical inside. Could we combine them in 1 general method usinf if , else if?

General method created

File: src/test/java/utils/ProjectConstants.java [hunk 4]
```diff
@@ -13,4 +13,6 @@ public class ProjectConstants {
     public static final String FORMS_URL = BASE_URL + "forms";
     public static final String PRACTICE_FORMS_HEADER = "Practice Form";
     public static final String PRACTICE_FORMS_URL = BASE_URL + "automation-practice-form";
+    public static final String SALT_UPPER_AND_LOWER_LETTERS = "AaBbCcDdEeFfGgHhIiJjKkLlMmNnOoPpQqRrSsTtUuVvWwXxYyZz";
```
Comment (KristinaPereselkina) [inline src/test/java/utils/ProjectConstants.java:16]: please not hardcode change to regex . for example /^[a-z]+$ (it contains upper and lower cases of alphabet)

File: src/test/java/utils/ProjectConstants.java [hunk 5]
```diff
@@ -13,4 +13,6 @@ public class ProjectConstants {
...
+    public static final String SALT_NUMBERS = "1234567890";
```
Comment (KristinaPereselkina) [inline src/test/java/utils/ProjectConstants.java:17]: the same, no hardcode. regex but for numbers

File: src/test/java/utils/ProjectConstants.java (hunk 4 above)
Comment (KristinaPereselkina) [reply src/test/java/utils/ProjectConstants.java:16]: https://regex101.com/ 
you can verify it here

**Review by KristinaPereselkina — CHANGES_REQUESTED** [review]

File: src/test/java/pages/BasePage.java [hunk 6]
```diff
@@ -428,5 +438,83 @@ protected boolean isAlertDisplayed() {
...
+            int index = (int) (rnd.nextFloat() * SALTCHARS.length());
+            salt.append(SALTCHARS.charAt(index));
+        }
+        String saltStr = salt.toString();
+        return saltStr;
+
+    }
+
+    protected String getSaltStringAge() {
+        String SALTCHARS = "1234567890";
```
[the remark this answers is not in the document]
Comment (yuriiormson) [reply src/test/java/pages/BasePage.java]: Where are you find it ?

**Review by KristinaPereselkina — APPROVED** [review]

**Review by KristinaPereselkina — APPROVED** [review]

**Review by KristinaPereselkina — APPROVED** [review]

File: src/test/java/pages/BasePage.java (hunk 1 above)
Comment (KristinaPereselkina) [reply src/test/java/pages/BasePage.java]: resolved

**Review by KristinaPereselkina — APPROVED** [review]

**Pull request merged**